### DISCUSSION TITLE
Labor für 5590 / FOS 7.50 für 7583

### DISCUSSION THREAD

**Discussion by maverick3005** [Freetz-NG/freetz-ng#737] [Unerfüllte Wünsche]
Könnten wir für die 5590 zu Testzwecken bitte mal das aktuelle Labor dem Freetz hinzufügen ?
Für die 7583 steht jetzt auch eine 7.50 final zur Verfügung.

Noch ein kleiner Hinweis zum "Image-Packing" der aktuellen 5590 7.30 FOS:

Dort ist ein neuer Bootloader mit enthalten welcher das Image einiges grösser macht.
Durch das Recovery zur 7.30 wird dieses Update bei der 5590 ebenso durchgeführt.

Meiner Meinung nach sollten wir dieses "Dauerupdate", welches ein einzelnes Binary ist, bei den Freetz-Images
entfernen, ebenso den dazugehörigen Eintrag im Updatescript. Das "dauerflaschen" inkl. dem Bootloader würde bei Stromausfall
doch ein gewisses "Briefbeschwerer"-Risiko darstellen ? - Es reicht wohl wenn das durch das Recovery bereits einmal upgedated
wurde, alternativ durch das einspielen des Originalimages beim Umstieg von 7.29->7.30

Mag ja sein, das ein mehrfaches Flashen unterbleibt sofern bereits der Bootloader auf aktuellem Stand ist, jedoch
ist das ebenso mitschleppen in Freetz-Images irgendwie sinnfrei dauerhaft...?

**Comment by fda77** [comment]
Ich kann aktuelle keine neueren Fos einbauen, die VM ist voll. Da Freetz open-source ist kann das aber jeder.
Und wenn du ger eine neue Option möchtest, mach es doch.
Ich kann mir nicht vorstellen (aber auch nicht geprüft) dass avm den bl ständig mit dem gleichen überschreibt

**Reply by maverick3005** [reply]
naja, dachte auch eher an die config scripte, die Labor 5590 ueber menuconfig ueberhaupt wie bei anderen 7.39ern erst mal in der Auswahl zu haben grundsätzlich und die entsprechenden download urls + checksum mit einzufügen.

Das mit dem nicht "ständig neu überschreiben" gehe ich im Normalfall ebenso von aus. Da wir aber AVM ja nun kennen und man in der Regel von einem "Basic-Stock image" ja dann freetzt normal, wollte ich nur anregen das wir dann in den freetz-images selber dieses update-binary nicht ebenso permanent weiterhin mit uns mitschleppen... Die images werden halt so wieder kleiner und wir schliessen dann ggfs. doch Risiken und Nebenwirkungen mit aus von Anfang an ;-)

Übrigens entfernt das Recovery für die 5590 NICHT die manuell im Datenbereich hinterlegten "fit-images" für den Fiberteil. Die Datenpartition bleibt weitestgehend nach einem Recovery weiterhin erhalten. Ist nicht unbedingt ein Problem, wollte nur anmerken das mit dem recovery kein 100%er Factory-Zustand tatsächlich wieder vollständig hergestellt wird.

**Reply by fda77** [reply]
> naja, dachte auch eher an die config scripte, die Labor 5590 ueber menuconfig ueberhaupt wie bei anderen 7.39ern erst mal in der Auswahl zu haben grundsätzlich und die entsprechenden download urls + checksum mit einzufügen.

Nur zu. Was ich dazu schrieb siehst du oben in https://github.com/Freetz-NG/freetz-ng/discussions/737#discussioncomment-4942389

> Übrigens entfernt das Recovery für die 5590 NICHT [..] Datenbereich  [..]

Das dürfte ein Fehler von AVM sein, bei anderen wie 7490 hab ich da schonmal aus versehen Dateien gelöscht. Ruf AVM an oder schreib ne Mail

> Die images werden halt so wieder kleiner  

Und? Das ist egal. Wenn die sie besonders klein haben willst nimm xz

> wir schliessen dann ggfs. doch Risiken und Nebenwirkungen mit aus von Anfang an ;-)

Mir fallen keine ein. Es entstehen dadurch eher welche

**Reply by maverick3005** [reply]
AVM hatte ich anlässlich des nicht sauber gecleanten "internen Speichers" und dem Cleaning schon ne Info gesendet.

Ich kenne das nämlich von allen anderen Boxen betreffs des TFFS-Cleanings auch nur, das diese Bereiche dann auch wirklich
vollständig clean sind. Bei der 5590 verbleibt jedoch in jedem Fall das manuell angelegte fit-image Dir samt Inhalt.
Aller Wahrscheinlichkeit nach dürften auch die freetz-Nodes ggfs. ebenso nicht restlos entfernt werden, das muss ich mir die Tage auch nochmal ansehen. Da machts dann wohl auch Sinn das mit dem bekannten Shell-script manuell sicherheitshalber noch zu löschen, wenn man von Factory -> freetz einmal initial sauber durchstarten will im Moment.

Eine simple lokale Erweiterung auf 7.39-Labor und einfache Anpassung auf die entsprechende Download-URL, nehme ich mal an, wird wohl sicherlich nicht ausreichen, da hier wohl auch noch das "7.50er" Start-Up-Geraffel, multid-Problematiken, anderer Kernel, keine Sourcen, usw. auch noch zu berücksichtigen wäre... gehe ich zumindest mal von aus das da noch ein paar Sachen mehr zu beachten wären.

Der Fehler mit dem Recovery und dem nicht vollständigen cleaning des TFFS-Images Bereichs im Datenbereich hat zumindest einen Vorteil:

Nan kann mehrere ausgepackte fit-images auf der Box im internen Speicher "aufbewahren" und kann sich das scp/ftp zum hochladen bei verschiedenen Versionen dann ein wenig ersparen. lol

**Comment by zackmuc** [comment]
Und wie baut man FOS ein?

**Reply by maverick3005** [reply]
Pauschal tät ich sagen, als Basis von einer anderen ähnlichen Box die auf gcc-5 toolchain basiert und 7.39 Labor unterstützt etwas copy & paste übernehmen und daraus dann "untested" 7.39 labor für die 5590 hinzufügen. Wg. dem Kernel der da drin ist müsste ich selber auch das Image erst einmal zerlegen und schaun was sich da gegenüber der 7.30 Final alles noch so geändert hat.

Das wirklich einzig Interessante, laut dem Changelog zur Labor wären die "verbesserte Kompatibilität" zu bestimten GPON-Gegenstellen.
Die Homeautomation und angeblichen Mesh-Verbesserungen, die allgemein bekannt sind von den anderen Boxen, würde ich da eher als nebensächlich ansehen. Auch sollten mehr ISP-Basis-Profile für kleinere, neue FTTH-Provider mehr integriert sein, wobei mir auch hier schon aufgefallen ist, das die Einstellung "anderer Anbieter "immer noch am sinnvollsten ist, da z.b. bei der Deutschen Glasfaser im Profil sich der Down/Upstream, VLAN-IDs und die ONT-ID (cloning...) via Menu dann nicht mehr confen lassen. Da kann man also bei Manuellem Profil weiterhin bleiben und alles vollständig weiterhin von Hand anlegen.

Im Bereich TR-69 ist mit aufgefallen, das beim Branding AVM und AVME im AVM-subordner 2 Certificate für 2 Deutsche ISPs mehr vorhanden sind. Die verfügbaren Boxen werden jedoch mit dem branding AVME ausgeliefert. Ich meine mich zu erinnern, das im AVME-config Ordner die Certificate für Telekom Deutschland und noch einem weiteren Provider für die TR69 autoconfig fehlen:

Das Cert "/freetz-ng/build/modified/filesystem/etc/default.Fritz_Box_HW272/avm/tr069-default-iad-fritzchen.telekom.de.pem" ist nur im "AVM"-Ordner vorhanden.

Beim kurzen Vergleich wäre es wohl einfacher sofern man auf AVME-Branding sein Image baut, die fehlenden certs aus dem AVM-Ordner mit den in AVME Ordner zu kopieren, oder den freetz symlink zu einem gemeinamen Ordner zu entfernen und stets den vollständigen Inhalt für AVM und AVME zur Verfügung zu haben und ggfs. das Branding beim freetz Image erstellen auf AVM "forced" zu überschreiben.

Frage: Werden eigentlich von Providern irgendwo auch 5590 mit AVM-Branding von Anfang an ausgeliefert oder gibt es ausschliesslich AVME branded Boxen im freien Handel UND von den ISPS ? Hat da jemand schon weitere Infos ?

Das oben genannte Certificate soll eigentlich für Telekom-Kunden relevant sein um deren Remote-Access wie bei den Speedports
ueblich, auch auf Fritzboxen nutzbar zumachen, Stichwort "ISP-Management, selbstständige Ersteinrichtung inkl. aller
SIP-Daten, usw...." (EasySupport laut AVM Changelog). 

Mein Stand ist aktuell der, das ausserhalb Deutschlands die FTTH-ISPs die 5590 als Kundengerät regulär mit anbieten, in Deutschland sind alle Anbieter leider immer noch Fans Ihrer ONT-Modems mit Kuperport nach intern. AVM/Fiber wird wie der Teufel ueberall noch gemieden und eher zwangsweise wg. Routerfreiheit "Geduldet" - ohne Support seitens des Anbieters.

**Comment by NDiGH** [comment]
> Dort ist ein neuer Bootloader mit enthalten welcher das Image einiges grösser macht. Durch das Recovery zur 7.30 wird dieses Update bei der 5590 ebenso durchgeführt.

Wäre mir neu. M.W.n. wird dieses Bootloader-Update vom AVM Recovery-Tool nicht durchgeführt.


> Meiner Meinung nach sollten wir dieses "Dauerupdate", welches ein einzelnes Binary ist, bei den Freetz-Images entfernen, ebenso den dazugehörigen Eintrag im Updatescript. Das "dauerflaschen" inkl. dem Bootloader würde bei Stromausfall doch ein gewisses "Briefbeschwerer"-Risiko darstellen ?

Ein "Dauerupdate" wäre mir auch neu. Soweit ich weiß wird zuvor geprüft ob ein Bootloader-Update notwendig ist. Und wenn nicht, wird dieses auch nicht durchgeführt.

Ich plädiere jedenfalls dafür, dass das Bootloader-Update nicht aus den Images entfernt wird. Zumindest nicht standardmäßig. Denn normalerweise habe ich nicht vor extra dafür zwischendurch mal ein originales Firmware-Image von AVM zu installieren oder gar ein Recovery durchzuführen.

**Reply by maverick3005** [reply]
Ja, so gesehen, wenn man z.b. unter freetz direkt von 7.29 auf > 7.30+ oder Labor geht macht das schon Sinn, das Update drin zu lassen. Haste wohl recht ;-)

**Comment by fda77** [comment]
Solang https://github.com/Freetz-NG/freetz-ng/actions/workflows/generate_img.yml nicht funktioniert sind mir neue FOS zu umständlich einzubauen

**Reply by maverick3005** [reply]
@fda77 auch wenn das "generate_img.yml" derzeit nicht so hinhaut wie Du magst.

Könntest Du evtl. zumindest das "Framework" für das 5590-Labor -UNTESTED- einmal mit Download-URL, etc. schon mal hinzufügen ?
Ich würde das gerne alles lokal dann mal durchtesten.

**Reply by fda77** [reply]
Die generate_img action läuft wieder, der Test liegt an dir ;-)
